great_expectations_provider/operators/great_expectations.py

```diff
@@ -243,7 +243,11 @@ def make_connection_string(self) -> str:
                 odbc_connector = "mssql+pyodbc"
             uri_string = f"{odbc_connector}://{self.conn.login}:{self.conn.password}@{self.conn.host}:{self.conn.port}/{schema}"  # noqa
         elif conn_type == "snowflake":
-            uri_string = f"snowflake://{self.conn.login}:{self.conn.password}@{self.conn.extra_dejson['extra__snowflake__account']}.{self.conn.extra_dejson['extra__snowflake__region']}/{self.conn.extra_dejson['extra__snowflake__database']}/{schema}?warehouse={self.conn.extra_dejson['extra__snowflake__warehouse']}&role={self.conn.extra_dejson['extra__snowflake__role']}"  # noqa
+            try:  # auto generated connection kwargs Snowflake provider >=4.0.0
+                uri_string = f"snowflake://{self.conn.login}:{self.conn.password}@{self.conn.extra_dejson['account']}.{self.conn.extra_dejson['region']}/{self.conn.extra_dejson['database']}/{schema}?warehouse={self.conn.extra_dejson['warehouse']}&role={self.conn.extra_dejson['role']}"  # noqa
+            except KeyError:  # auto generated connection kwargs Snowflake provider < 4.0.0
```

Comment:
I think a more direct approach to check package versions here would be better, something like the answer here:
https://stackoverflow.com/a/32965521

Reply:
That makes sense, yes. Changed it :)

One issue: referencing the snowflake package is breaking the unit testing. I tried using the package import and then .__version__ but the snowflake package does not seem to have that as an attribute. 

The code itself works fine. Just the test is understandably unhappy.

Reply:
Maybe @kaxil has an idea of how packages can be brought in for testing? I haven't done anything like that before afaik

Reply:
Ok I was wrong about the package approach -- totally forgot about `.get()` even though I used it today 😓